app/controllers/orgs/rosters_controller/google_classroom_dependency.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+module Orgs
+  class RostersController
+    before_action :ensure_google_classroom_roster_import_is_enabled, only: %i[
+      import_from_google_classroom sync_google_classroom
+    ]
+    before_action :authorize_google_classroom, only: %i[import_from_google_classroom sync_google_classroom]
+    before_action :ensure_google_classroom_is_linked, only: %i[import_from_google_classroom sync_google_classroom]
+    before_action :set_google_classroom, only: %i[import_from_google_classroom sync_google_classroom]
+    before_action :ensure_no_lti_configuration, only: :import_from_google_classroom
+    before_action :google_classroom_ensure_no_roster, only: :import_from_google_classroom
+
+    def import_from_google_classroom
+      students = list_google_classroom_students
+      return unless students
+
+      if students.blank?
+        flash[:warning] = "No students were found in your Google Classroom. Please add students and try again."
+        redirect_to roster_path(current_organization)
+      else
+        add_google_classroom_students(students)
+      end
+    end
+
+    def sync_google_classroom
+      unless current_organization.google_course_id
+        flash[:error] = "No Google Classroom has been linked. Please link Google Classroom."
+        return
+      end
+
+      latest_students = list_google_classroom_students
+      latest_student_ids = latest_students.collect(&:user_id)
+      current_student_ids = current_roster.roster_entries.collect(&:google_user_id)
```

Comment:
can we replace this by something like this:
`current_student_ids = current_roster.roster_entries.pluck(&:google_user_id)`
This will result in a `SELECT roster_entries.google_user_id` instead of `SELECT roster_entries.*`.
This can be particularly be helpful if the number of roster entries is large.

Reply:
Oh I like that idea a lot. I always forget that is something you can do in rails.